src/engine/errors.rs

```diff
@@ -0,0 +1,60 @@
+use std::{error::Error, fmt::Display};
+
+use tonic::codegen::http::StatusCode;
+
+#[derive(Debug)]
+pub struct EngineError {
+    error: anyhow::Error,
+    pub code: StatusCode,
+}
```

Comment:
Can you add some documentation to this struct?

Reply:
Oh yeah I forgot about that, I'll add some rn

Reply:
done